src/CognitiveServices/CognitiveServices/CognitiveServicesAccount/NewAzureCognitiveServicesAccount.cs

```diff
@@ -118,23 +119,36 @@ public override void ExecuteCmdlet()
                 if (ShouldProcess(
                     Name, string.Format(CultureInfo.CurrentCulture, Resources.NewAccount_ProcessMessage, Name, Type, SkuName, Location)))
                 {
-                    if (Force.IsPresent)
+                    if (Type.StartsWith("Bing.", StringComparison.InvariantCultureIgnoreCase))
                     {
-                        WriteWarning(Resources.NewAccount_Notice);
-                    }
-                    else
-                    {
-                        bool yesToAll = false, noToAll = false;
-                        if (!ShouldContinue(Resources.NewAccount_Notice, "Notice", true, ref yesToAll, ref noToAll))
+                        if (Force.IsPresent)
+                        {
+                            WriteWarning(Resources.NewAccount_Notice);
+                        }
+                        else
                         {
-                            return;
+                            bool yesToAll = false, noToAll = false;
+                            if (!ShouldContinue(Resources.NewAccount_Notice, "Notice", true, ref yesToAll, ref noToAll))
+                            {
+                                return;
+                            }
                         }
                     }
+                    try
+                    {
+                        CognitiveServicesAccount createAccountResponse = CognitiveServicesClient.Accounts.Create(
+                                        ResourceGroupName,
+                                        Name,
+                                        createParameters);
+                    }
+                    catch (Exception ex)
+                    {
+                        // Give users a specific message says `Failed to create Cognitive Services account.`
+                        // Details should able be found in the exception.
+                        WriteError(new ErrorRecord(new Exception("Failed to create Cognitive Services account.", ex), string.Empty, ErrorCategory.NotSpecified, null));
```

Comment:
@yangyuan any reason that you aren't just throwing the new `Exception` here?

Reply:
No specific reason. In here, `ErrorRecord` doesn't contribute extra information, so it's just for consistency.
Do you recommend to change to simple `Exception`?

Reply:
@yangyuan we advise just throwing an `Exception` in this case